### DISCUSSION TITLE
recommendations: same ID in positive/negative

### DISCUSSION THREAD

**Discussion by dnalor** [qdrant/qdrant#5645] [Q&A] [answered]
What happens if the same point ID is in both the positive and negative parameter for a recommendation request?
- Cancel each other out?
- Detected by qdrant?
- ???

(Background: Music recommendation system. Recommended IDs are saved in session and used as positive when requesting next tracks. User can mark tracks as "disliked", these will be added to the negative IDs for the following recommendation requests)

**Comment by timvisee** [comment, marked as answer]
Providing a point as positive and negative example will make them cancel each other out.

These two searches will give the same result:

```json
POST /collections/benchmark/points/query
{
  "query": {
    "recommend": {
      "positive": [2],
      "negative": []
    }
  }
}
```

```json
POST /collections/benchmark/points/query
{
  "query": {
    "recommend": {
      "positive": [1, 2],
      "negative": [1]
    }
  }
}
```

**Reply by dnalor** [reply]
Thanks for the reply.
This makes sense and likely is the expected behavior for most people (although for my specific use case it means I'll need to de-duplicate the positive/negative IDs before making the request. Admittedly, if qdrant would do this internally, this would probably be too much magic)

**Reply by timvisee** [reply]
> Admittedly, if qdrant would do this internally, this would probably be too much magic

Yeah we don't want to do these things by default because this way we still give you freedom to implement it either way :smile:
